### PR TITLE
samples: timer wakeup

### PR DESCRIPTION
**Pull request by wilu-nordic**

### CODE REVIEW

File: samples/timer_wakeup/src/main.c [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	};
+
+	/* Set the TIMER configuration. */
+	npmx_timer_config_set(m_timer_inst, &timer_config);
+
+	/* Set LED to be controlled by the host and eventually turn it off. */
+	npmx_led_mode_set(npmx_led_get(npmx_inst, LED_IDX), NPMX_LED_MODE_HOST);
+	npmx_led_state_set(npmx_led_get(npmx_inst, LED_IDX), false);
+
+	/* Save compare value data to internal timer's registers. */
```
Comment (annakielar) [inline samples/timer_wakeup/src/main.c:173]:
```suggestion
	/* Save the compare value data to internal timer's registers. */
```

File: samples/timer_wakeup/README.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the TIMER and SHIPHLD driver to do the following:
+
+  * Configure SHIPHLD button.
+  * Register handler to be called when the button is pressed.
+  * Configure TIMER in wake-up mode.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/timer_wakeup/README.rst:28]: Periphery

File: samples/timer_wakeup/README.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Press SW1 button (SHPHLD) to enter the hibernate mode.
+#. Wait for ~8 seconds until device wakes up.
+#. LEDs is used to indicate when the PMIC wakes up.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/timer_wakeup/README.rst:91]: The LED is on when PMIC is up.

File: samples/timer_wakeup/README.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Press the SW1 (SHPHLD) button to enter the hibernate mode.
+#. Wait for ~8 seconds until device wakes up.
+
+LEDs are used to indicate when the PMIC wakes up.
```
Comment (annakielar) [inline samples/timer_wakeup/README.rst:92]: Here we should have a result of testing actions, so the sentence is in a correct place but we need to change its structure, for example:
```suggestion
LEDs will start blinking when the PMIC wakes up.
```

File: samples/timer_wakeup/src/main.c [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+npmx_ship_t *m_ship_inst;
+
+/** @brief Declaration of the pointer to the TIMER driver instance. */
+npmx_timer_t *m_timer_inst;
+
+/** @brief Variable filled in @ref button_callback() when the TIMER starts working. */
+static volatile uint32_t start_time;
+
+/**
+ * @brief Function for blinking LEDs.
```
Comment (annakielar) [inline samples/timer_wakeup/src/main.c:46]:
```suggestion
 * @brief Function for enabling LEDs to blink.
```
Is it correct?

**Review by annakielar — CHANGES_REQUESTED** [review]

File: samples/timer_wakeup/README.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
+.. _timer_wake_up_sample:
+
+Wake-up Timer
+#############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Wake-up Timer sample demonstrates how to use the PMIC TIMER periphery and SHIPHLD button with npmx drivers.
```
Comment (annakielar) [inline samples/timer_wakeup/README.rst:10]:
```suggestion
The Wake-up Timer sample demonstrates how to use the PMIC TIMER periphery and the SHIPHLD button with npmx drivers.
```

**Review by annakielar — CHANGES_REQUESTED** [review]

File: samples/timer_wakeup/README.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Press the SW1 (SHPHLD) button to enter the hibernate mode.
+#. Wait for ~8 seconds until device wakes up.
```
Comment (annakielar) [inline samples/timer_wakeup/README.rst:90]:
```suggestion
#. Wait for ~8 seconds until the device wakes up.
```

File: samples/timer_wakeup/README.rst [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+LED is on when the PMIC wakes up.
```
Comment (annakielar) [inline samples/timer_wakeup/README.rst:92]:
```suggestion
When the PMIC wakes up, the LED will turn on.
```

**Review by annakielar — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

File: samples/timer_wakeup/README.rst [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Press the SW1 (SHPHLD) button to enter the hibernate mode.
```
Comment (masz-nordic) [inline samples/timer_wakeup/README.rst:89]: `hibernate mode` or `hibernation mode`?

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**